### PR TITLE
refactor(validate file transfer)

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

part of https://github.com/Clinical-Genomics/add-new-tech/issues/100

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/services/validate_file_transfer_service/validate_file_transfer_service.py [hunk 1]
```diff
@@ -5,34 +5,49 @@
...
+    ):
+        """Validate all files listed in the manifest are present in the directory tree."""
+        files_in_manifest: list[str] = self._get_files_in_manifest(
+            manifest_file=manifest_file, manifest_file_format=manifest_file_format
+        )
+        if not self._are_all_files_present(
+            files_to_validate=files_in_manifest, source_dir=source_dir
+        ):
+            raise FileNotFoundError(
+                f"Not all files listed in the manifest file {manifest_file} are not present in the directory tree {source_dir}"
```
Comment (islean) [inline cg/services/validate_file_transfer_service/validate_file_transfer_service.py:21]:
```suggestion
                f"Not all files listed in the manifest file {manifest_file} are present in the directory tree {source_dir}"
```

File: cg/services/validate_file_transfer_service/validate_file_transfer_service.py [hunk 2]
```diff
@@ -5,34 +5,49 @@
...
         """Get the content of the manifest file."""
         file_reader = ReadFile()
         return file_reader.get_content_from_file(
             file_format=manifest_file_format, file_path=manifest_file
         )
 
     @staticmethod
-    def is_valid_path(line: str):
+    def _is_valid_path(line: str):
         return "/" in line
```
Comment (islean) [inline cg/services/validate_file_transfer_service/validate_file_transfer_service.py:35]: Out of scope probably but this method seems off to me. Is there not a better check to do?

Comment (islean) [reply cg/services/validate_file_transfer_service/validate_file_transfer_service.py:35]: Like a Path.is_dir()?

**Review by islean — APPROVED** [review]
Looks good!

File: cg/services/validate_file_transfer_service/validate_file_transfer_service.py (hunk 2 above)
Comment (ChrOertlin) [reply cg/services/validate_file_transfer_service/validate_file_transfer_service.py:35]: Yes I agree, I will have a look at this in a different pr.

**Comment by ChrOertlin** [comment]
ignoring coveralls

**Pull request merged**